src/main/webapp/authorization.js

```diff
@@ -1,56 +1,115 @@
+   */
+  getUser() {
+    return this.firebase.auth().currentUser;
+  }
 
-/**
- * Function called when clicking the Login/Logout button.
- */
-function toggleSignIn() {
-  const firebase = getFirebase();
-  if (!firebase.auth().currentUser) {
-    let provider = new firebase.auth.GithubAuthProvider();
-    firebase.auth().signInWithPopup(provider).then(function(result) {
-      // This gives you a GitHub Access Token.
-      // You can use it to access the GitHub API.
-      const token = result.credential.accessToken; //eslint-disable-line
-      // The signed-in user info.
-      const user = result.user; //eslint-disable-line
-    }).catch(function(error) {
-      // Handle Errors here.
-      console.error(error);
-      // Change this in future iterations
-      alert(error.message);
-    });
-  } else {
-    // if logged in, sign out
-    firebase.auth().signOut();
+  /**
+   * Get firebase instance used with this object
+   * @return {Firebase} firebase instance
+   */
+  getFirebase() {
+    return this.firebase;
+  }
+
+  /**
+   * Function called when authorizing user to verify them as a github user.
+   * This function doubles as an opt out when the current user wants to leave
+   * our site for good.
+   *
+   * Side Effects: when authorizing a user, it sets the current objects token
+   * to the returned Github API accessToken if a token is returned.
+   *
+   * NOTE: this function is asynchronous and should be used with an await
+   *
+   * @return {string} the GitHub API access token, null if an error occurs or
+   * the given user is no longer athorized.
+   */
+  async toggleAuthorization() {
+    const firebase = this.firebase;
+    if (!firebase.auth().currentUser) {
+      try {
+        let provider = new firebase.auth.GithubAuthProvider();
+        let authorizationResults =
+          await firebase.auth().signInWithPopup(provider);
+        // This gives you a GitHub Access Token.
+        // You can use it to access the GitHub API.
+        this.token = await authorizationResults.credential.accessToken;
+      } catch (error) {
+        // Handle Errors here.
+        console.error(error);
+        // TODO: Change this in future iterations
+        alert(error.message);
+        this.token = null;
+      }
+    } else {
+      // if logged in, sign out
+      firebase.auth().signOut();
+      this.token = null;
+    }
+    return this.token;
+  }
+
+  /**
+   * Initialize firebase if firebase is not already initialized. This method can
+   * be called as many times as we need without any side effects.
+   *
+   * NOTE: This is not an instance function and will not be exported.
+   */
+  static initializeFirebase() {
+    if (firebase.apps.length === 0) { // eslint-disable-line
+      firebase.initializeApp(firebaseConfig); // eslint-disable-line
+    }
   }
 }
 
-export {
-  initializeFirebase,
-  toggleSignIn,
-  getFirebase,
-};
+export let githubAuthorizer = new GithubAuthorizer();
```

Comment:
Not necessary but a cool tip:
```suggestion
export default new GithubAuthorizer();
```

This allows for us to more cleanly import the module without needing curly braces in the import statement. It also means we don't need to strictly match the naming of the variable being exported. For comparison: 
```javascript
import GitHubAuthorizer from './file.js`; // export default
import { githubAuthorizer } from './file.js'; // not export default
```

You can learn more in [this article](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#Using_the_default_export).

Reply:
We're not supposed to use default exports according to the style guide.

Reply:
I was really sad about that, it seemed like a good way to package the class instance

Reply:
Oh crap really? I didn't know that and I have been using them often oops.